tools/rpkg/src/statement.cpp

```diff
@@ -572,6 +590,20 @@ static void transform(Vector &src_vec, SEXP &dest, idx_t dest_offset, idx_t n, b
 		SET_CLASS(dest, RStrings::get().factor_str);
 		break;
 	}
+	case LogicalTypeId::UUID: {
+		auto src_ptr = FlatVector::GetData<hugeint_t>(src_vec);
+		auto &mask = FlatVector::Validity(src_vec);
+		for (size_t row_idx = 0; row_idx < n; row_idx++) {
+			if (!mask.RowIsValid(row_idx)) {
+				SET_STRING_ELT(dest, dest_offset + row_idx, NA_STRING);
+			} else {
+				char uuid_buf[UUID::STRING_SIZE];
```

Comment:
Are we risking a buffer overflow here?

```suggestion
				char uuid_buf[UUID::STRING_SIZE + 1];
```

Reply:
DuckDB doesn't work with null-terminated strings internally, and since `Rf_mkCharLen` is used afterwards the buffer should be of sufficient size. `UUID::ToString` will write exactly `UUID::STRING_SIZE` characters so the code looks correct to me.